### ISSUE TITLE
FIlter pruning with LeGR

### ISSUE THREAD

**Issue by Hyunseok-Kim0** [openvinotoolkit/nncf#1689]
I am trying to apply filter pruning with LeGR to yolov8 model. It was able to run the pruning itself, but I have few questions about the implementation and result.

1. In train_step during pruning, NNCFNetwork is trained for few epochs. However, in the early step of pruning, the model already shows almost 0 map even before training is started. is this normal for LeGR? I know normal filter pruning would not work like that, but not sure for LeGR.

2. In first epoch of train_step, I observed the weight and other things are same with original model. However, the inference output of NNCFNetwork is totally different with original one. The only difference between two models is pruning mask inserted in the model. Is the output difference caused by the mask? 

3. There are few examples about LeGR in the branch https://github.com/mkaglins/nncf_pytorch/tree/mkaglins/RL_experiments. Unfortunately those examples are too old, so it looks sample codes are not fit with the current version. Is there any sample code for current version of LeGR?

4. As mentioned in issues like #1038, pruned pytorch model have same size with original one. Will the model size decreased based on the pruning mask if I export the model to ONNX?

5. Training results during each generation looks completely identical. It looks pruning mask does not affect anything to output of network, which is weird since the output itself is different from original network. https://github.com/openvinotoolkit/nncf/blob/35e21e96f27e51e3dfcd32882b32e8d5312ec983/nncf/torch/pruning/filter_pruning/global_ranking/evolutionary_optimization.py#L101-L110 In line number 106, random seed is reassigned every time `_predict_action` called. Due to the line, actions within population number always have same value. Is this thing intended or just a bug? Even after I commented out that line, training result and reward are always same.

**Comment by daniil-lyakhov** [comment]
Hello, @Hyunseok-Kim0! Sorry for a such a long delay, I've lost your questions in my mail and found it only now. Let me try to answer some of your questions:

1-2) Filter pruning is applying pruning masks to the model during `create_compressed_model`. Since masks operations are applied, they are zeroing some channels of convolution and linear operations, so despite the fact that all model parameters are the same, activations computation is modified by zero channels of some conv and linear layers.
3) You can try the [Torch classification sample](https://github.com/openvinotoolkit/nncf/tree/develop/examples/torch/classification) with a [LEGR config](https://github.com/openvinotoolkit/nncf/blob/develop/examples/torch/classification/configs/pruning/resnet18_imagenet_pruning_learned_ranking.json).
4) Unfortunately no, the only way to remove parameters from the model is to use Openvino pruning transformation as it described [here](https://github.com/openvinotoolkit/nncf/issues/1038)
5) The answer for this question requires some experiments and investigations, please contact me again if you really need the answer for this question

Hope that will help! Please don't hesitate to contact me directly by e-mail in case of missed messages

**Comment by MaximProshin** [comment]
Issue closed as no more feedback for a long time period. Feel free to re-open it if needed.
